Add a new HAFS regression test with Thompson MP and GFDL surface layer schemes and Corrected S4 template regression testing job card

ChunxiZhang-NOAA opened the pull request:
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

Created a new regression test for the newly added HAFS physics suite, e.g., suite_FV3_HAFS_v0_thompson_tedmf_gfdlsf.xml (Note that the other newly added suite, e.g. suite_FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst.xml is not included in this regression test because this one is very similar to the first one). 

Corrected S4 template regression testing job card

Contributors: @ChunxiZhang-NOAA, @BinLiu-NOAA, @DavidHuber-NOAA

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes https://github.com/ufs-community/ufs-weather-model/issues/987
- fixes https://github.com/ufs-community/ufs-weather-model/issues/900
- fixes https://github.com/ufs-community/ufs-weather-model/issues/991

## Testing

Regression tests related to the HAFS were conducted on Hera.intel. All tests were passed except the newly added 'hafs_regional_atm_thompson_gfdlsf' due to missing baseline.

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/fv3atm/pull/459

climbfuji commented on tests/parm/field_table/field_table_hafs_thompson:
```diff
@@ -0,0 +1,65 @@
+# added by FRE: sphum must be present in atmos
```
What is the difference between this file and `https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/field_table/field_table_thompson_aero_tke`?

ChunxiZhang-NOAA replied:
field_table_hafs_thompson is the same as field_table_thompson_noaero_tke. I just want to hold a spot for future changes of field_table_hafs_thompson. Note that field_table_hafs is the same as field_table_gfsv16.

BinLiu-NOAA commented on tests/rt.conf:
```diff
@@ -142,8 +142,9 @@ RUN     | control_wam_debug
 ###################################################################################################################################################################################
 #
-COMPILE | -DAPP=HAFSW -DCCPP_SUITES=FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_gfdlmp_tedmf_nonsst -D32BIT=ON                           | - wcoss2                                | fv3 |
+COMPILE | -DAPP=HAFSW -DCCPP_SUITES=FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_thompson_tedmf_gfdlsf,FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst,FV3_HAFS_v0_gfdlmp_tedmf_nonsst -D32BIT=ON                           | - wcoss2                                | fv3 |
```
@ChunxiZhang-NOAA, could you change this line into the following? Thanks!
COMPILE | -DAPP=HAFSW -DCCPP_SUITES=**FV3_HAFS_v0_gfdlmp_tedmf,FV3_HAFS_v0_gfdlmp_tedmf_nonsst,FV3_HAFS_v0_thompson_tedmf_gfdlsf,FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst** -D32BIT=ON                           | - wcoss2                                | fv3 |

ChunxiZhang-NOAA replied:
Changed as you suggested.

BinLiu-NOAA commented on tests/tests/hafs_regional_atm_thompson_gfdlsf:
```diff
@@ -0,0 +1,98 @@
+###############################################################################
+#
+#  HAFS regional atmosphere only
```
@ChunxiZhang-NOAA, Please change this line into:
HAFS regional atmosphere only using the Thompson microphysics and GFDL surface layer schemes
or something similar.

ChunxiZhang-NOAA replied:
Changed as you suggested

BinLiu-NOAA commented on tests/tests/hafs_regional_atm_thompson_gfdlsf:
```diff
@@ -0,0 +1,98 @@
+#
+###############################################################################
+
+export TEST_DESCR="Compare HAFS regional atmosphere only results with previous trunk version"
```
@ChunxiZhang-NOAA, also please update this line into the following or something similar. Thanks!
export TEST_DESCR="Compare HAFS regional atmosphere only results using the Thompson microphysics and GFDL surface layer schemes with previous trunk version"

ChunxiZhang-NOAA replied:
Changed to "Compare the results from HAFS regional atmosphere only using the Thompson microphysics scheme and GFDL surface layer scheme with previous trunk version".

binli2337 commented:
@ChunxiZhang-NOAA Please update your PR and add the changes from PR #992. Thanks.

ChunxiZhang-NOAA commented:
I combined David's change in PR#992 to this PR branch and synced my branch with the latest develop branch for both FV3 and ufs-weather-model level.  New baseline for this 'hafs_regional_atm_thompson_gfdlsf' regression test need to be added.

BinLiu-NOAA reviewed (approved):
Changes look good to me.

junwang-noaa commented on tests/parm/input_regional_hafs.nml.IN:
```diff
@@ -158,7 +158,12 @@
+       lradar         = .true.
+       ttendlim       = 0.004
+       sedi_semi      = .false.
+       ext_diag_thompson = .false.
```
Just to confirm those variables ltaerosol, lradar, ttendlim and sedi_semi are only used by Thompson, they have no impact on other MP schemes, right?

ChunxiZhang-NOAA replied:
Confirmed: those variables ltaerosol, lradar, ttendlim and sedi_semi are only used by Thompson, and they have no impact on other MP schemes.

DavidHuber-NOAA commented:
Looks good to me, also.  Thanks @ChunxiZhang-NOAA!

binli2337 commented:
@ChunxiZhang-NOAA The updated code cannot be compiled on Hera and Gaea.

binli2337 commented:
@ChunxiZhang-NOAA Please see error messages at /scratch1/NCEPDEV/stmp2/Bin.Li/FV3_RT/rt_19027/compile_001.

ChunxiZhang-NOAA commented:
Hi Bin, sorry for that. I found it is caused by a suite we added, e.g., FV3_HAFS_v0_thompson_tedmf_gfdlsf_nonsst. The regression test we added actually didn't use this suite. I will delete this suite and push the changes to my fork.

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/814264612/20220118194511/ufs-weather-model
/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220113/GNU
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
cheyenne-gnu-BL

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/814264612/20220118194612/ufs-weather-model
/glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220113/INTEL
 Exists already. It should not yet. Please delete.
Please make changes and add the following label back:
cheyenne-intel-BL

climbfuji commented:
Did you just want to verify against the baseline? It says 20220113 already exists.

> On Jan 18, 2022, at 7:47 PM, Brian Curtis ***@***.***> wrote:
> 
> 
> Automated RT Failure Notification
> Machine: cheyenne
> Compiler: intel
> Job: BL
> Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/814264612/20220118194612/ufs-weather-model
> /glade/p/ral/jntp/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220113/INTEL
> Exists already. It should not yet. Please delete.
> Please make changes and add the following label back:
> cheyenne-intel-BL
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ufs-community/ufs-weather-model/pull/995#issuecomment-1016027186>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB5C2RIBTH6J6KZBLRVQDHDUWYQ3DANCNFSM5LIR77MA>.
> Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
> You are receiving this because your review was requested.
>

junwang-noaa commented:
We just need to add one test hafs_regional_atm_thompson_gfdlsf's results to current baseline.

binli2337 commented:
@ChunxiZhang-NOAA @junwang-noaa All tests passed on Cray. RegressionTests_wcoss_cray.log is located at /gpfs/hps3/stmp/Bin.Li/ufs-weather-model/tests.
